File: Rock_paper_scissor.py
#import random module
import random

#game rule
print("Game rules:\n\tRock vs Paper => Paper Wins \n\tPaper vs Scissor => Scissor Wins \n\tRock vs Scissor => Rock Wins")
#array for the choices
choiceArray = ["Rock","Paper","Scissor"]
#To keep playing
while True:
    print("\nChoices are: \n\t1.Rock \n\t2.Paper \n\t3.Scissor")

    #Computer's choice
    print("\nComputer's turn")
    print("----------------------")
    print("Computer has done secretly. Obviously You're not able to see it now")
    computer_choice = random.randint(0,2)

    # Print user choice
    print("\nYour turn")
    print("--------------")
    user_choice = int(input("Enter your choice: "))-1
    while user_choice > 2 or user_choice < 0:
        user_choice = int(input("enter valid input: "))

    print("\n" + choiceArray[computer_choice] + " vs " + choiceArray[user_choice])

    if computer_choice == user_choice:
        print("\nDraw")
    elif ((computer_choice == 1 and user_choice == 2) or (computer_choice == 2 and user_choice == 3) or (computer_choice == 3 and user_choice == 1)):
        print(choiceArray[user_choice] + " wins. You Win")
    else:
        print(choiceArray[computer_choice] + " wins. Computer wins")

    print("\nDo you want to continue (Y/N)")
    answer=input()

    if (answer == "N" or answer == "n"):
        break

print("Thanks for playing")

File: rock_paper_scissor.rb

#game rule
puts("Game rule:\n\tRock vs Paper => Paper Win \n\tPaper vs Scissor => Scissor Win \n\tRock vs Scissor => Rock Win")

#For play again
while true  
    puts("\nChoices are: \n\t1.Rock \n\t2.paper \n\t3.Scissor")

    choice_names = ["Rock","Paper","Scissor"]

    #This will decide how a material wins agains other
    win = {'Rock':'Scissor','Scissor':'Paper','Paper':'Rock'}

    #Computer choice    
    puts("\nComputer's turn")
    puts("----------------------")
    puts("Computer has done secretly. Obviously You're not able to see it now")
    computer_choice = rand(1...3)
    

    # puts user choice
    puts("\nYour turn")
    puts("--------------")
    puts("Enter your choice: ")
    user_choice = gets().to_i
    while user_choice > 3 or user_choice < 1
        puts ("enter valid input:" )
        user_choice = gets().to_i
    end
    
    computer_material = choice_names[computer_choice-1]
    user_material = choice_names[user_choice-1]

    puts("\n" + computer_material + " vs " + user_material )

    if computer_material == user_material
        puts("\nDraw")
    elsif win[user_material] == computer_material
        puts(user_material + " wins. You Win")
    else
        puts(computer_material + " wins. Computer win")

        puts("\nDo you want to continue (Y/N)")
        answer = gets().chomp.to_s

        if answer.downcase.eql? "n"
            break
        end
    end
end

puts("Thanks for playing")